### PR TITLE
Re-enable Nimbus with Off the main thread I/O and non-blocking networking

### LINKED ISSUE #17086: [Bug] Large regression in MAIN/VIEW start up due to Nimbus network IO
On our FNPRMS testing dashboard:
image

We regressed from ~0.5s to ~1.4s. Let's find the root cause and fix it.

### Device information

* Android device: ?
* Fenix version: ?

---

This may be caused by the new strict mode violation https://github.com/mozilla-mobile/fenix/issues/16984 or the other network main thread issues we've seen with experiments. #16901 

Tracking bug for nimbus-sdk team: https://jira.mozilla.com/browse/SDK-157

### LINKED ISSUE #17143: [Bug] Crashes initializing Nimbus
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have the exact bug and steps to reproduce described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

## Steps to reproduce

Click "+" button to open new tab.

### Expected behavior

No crash and open new tab.

### Actual behavior

Crash when click "+" button to open new tab.

Report: https://crash-stats.mozilla.org/report/index/0d29e773-b26e-4361-b64b-a07c90201219

### Device information

* Android device: ASUS ZenFone 3 (Android 6.0.1)
* Fenix version:

### PR DESCRIPTION
**Pull request by jhugman**
Fixes [SDK-152][0].

In an effort to fix [SDK-157][1] this PR brings together: 

1. https://github.com/mozilla/uniffi-rs/pull/372 — allow Rust objects to be marked as Threadsafe, to indicate to uniffi not to lock access to one method at a time.
2. https://github.com/mozilla/nimbus-sdk/pull/86 — refactor Nimbus to use its own locking, instead of relying on uniffi. This also introduces an in-memory cache of enrollments.
3. https://github.com/mozilla/nimbus-sdk/pull/78 — split the long running, blocking and single threaded `updateExperiments` into a slow network `fetch` and a fast `apply` phase to re-calculate enrollments which writes to the database.
4. https://github.com/mozilla-mobile/android-components/pull/9394 — Move `fetch` calls and calls that mutate the internal state of Nimbus to two separate coroutine scopes.

Furthermore, this PR also: moves setup of the Nimbus object into its own file; starts using the new `fetch` and `apply` methods, and improves error resilience.

Once all of this lands, this fixes #17086. Indirectly, it fixes #17143.

[0]: https://jira.mozilla.com/browse/SDK-152
[1]: https://jira.mozilla.com/browse/SDK-157

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- ~[ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not~ Has no UI
- ~[ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.~ Has no UI.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by jhugman** [comment]
> IMPORTANT: Only turn this back on once the following issues are resolved:
>
> - https://github.com/mozilla-mobile/fenix/issues/17086: Calls to `getExperimentBranch` seem to block on `updateExperiments` causing a large performance regression loading the home screen.
> - https://github.com/mozilla-mobile/fenix/issues/17143: Despite having wrapped `getExperimentBranch`/`withExperiments` in a catch-allusers are still experiencing crashes.

#17143 is fixed by way of SDK-161 and https://github.com/mozilla/nimbus-sdk/pull/77 and extra try/catch protection around Nimbus setup.

File: app/src/main/java/org/mozilla/fenix/components/Analytics.kt [hunk 1]
```diff
@@ -8,17 +8,14 @@ import android.app.Application
 import android.app.PendingIntent
 import android.content.Context
 import android.content.Intent
-import android.net.Uri
-import android.os.StrictMode
+
```
Comment (travis79) [inline app/src/main/java/org/mozilla/fenix/components/Analytics.kt:11]: nit: blank line

**Review by travis79 — APPROVED** [review]
🎉

**Review by csadilek — APPROVED** [review]
We did a few round of manual testing, simulating slow networks and no network connection at all. Everything seems fine now :).

At this point we can land and should monitor Sentry and perf. results.

**Pull request merged**